enzyme/test/Enzyme/ReverseMode/blas/cblas_dgemm_col_mod1_split.ll

```diff
@@ -0,0 +1,142 @@
+  ret i32 0
+}
+
+declare void @llvm.memcpy.p0i8.p0i8.i64(i8* noalias nocapture writeonly, i8* noalias nocapture readonly, i64, i1)
+
+declare void @llvm.memset.p0i8.i64(i8* nocapture writeonly, i8, i64, i1)
+
+declare dso_local double @__enzyme_autodiff(i8*, double*, double*, double*, double*, double*, double*, double, double)
+
+;CHECK:define internal { double*, double* } @augmented_outer(double* %Afoo, double* %"Afoo'", double* %Bfoo, double* %"Bfoo'", double* %Cfoo, double* %"Cfoo'", double %alpha, double %beta) {
+;CHECK-NEXT:entry:
+;CHECK-NEXT:  %malloccall = tail call i8* @malloc(i64 mul (i64 ptrtoint (double* getelementptr (double, double* null, i32 1) to i64), i64 8))
+;CHECK-NEXT:  %0 = bitcast i8* %malloccall to double*
+;CHECK-NEXT:  %1 = bitcast double* %Afoo to i8*
+;CHECK-NEXT:  call void @llvm.memcpy.p0i8.p0i8.i32(i8* %malloccall, i8* %1, i32 8, i1 false)
+;CHECK-NEXT:  %malloccall2 = tail call i8* @malloc(i64 mul (i64 ptrtoint (double* getelementptr (double, double* null, i32 1) to i64), i64 6))
+;CHECK-NEXT:  %2 = bitcast i8* %malloccall2 to double*
+;CHECK-NEXT:  %3 = bitcast double* %Bfoo to i8*
+;CHECK-NEXT:  call void @llvm.memcpy.p0i8.p0i8.i32(i8* %malloccall2, i8* %3, i32 6, i1 false)
+;CHECK-NEXT:  %4 = insertvalue { double*, double* } undef, double* %0, 0
+;CHECK-NEXT:  %5 = insertvalue { double*, double* } %4, double* %2, 1
+;CHECK-NEXT:  call void @cblas_dgemm(i32 102, i32 111, i32 111, i32 4, i32 3, i32 2, double %alpha, double* nocapture readonly %Afoo, i32 4, double* nocapture readonly %Bfoo, i32 2, double %beta, double* %Cfoo, i32 4)
+;CHECK-NEXT:  ret { double*, double* } %5
+;CHECK-NEXT:}
+
+;CHECK:define internal { double, double } @diffeouter(double* %Afoo, double* %"Afoo'", double* %Bfoo, double* %"Bfoo'", double* %Cfoo, double* %"Cfoo'", double %alpha, double %beta, { double*, double* }
+;CHECK-NEXT:entry:
+;CHECK-NEXT:  %1 = extractvalue { double*, double* } %0, 0
+;CHECK-NEXT:  %2 = extractvalue { double*, double* } %0, 1
+;CHECK-NEXT:  call void @cblas_dgemm(i32 102, i32 111, i32 112, i32 4, i32 2, i32 3, double %alpha, double* nocapture readonly %"Cfoo'", i32 4, double* nocapture readonly %2, i32 3, double 1.000000e+00, double* %"Afoo'", i32 4)
+;CHECK-NEXT:  %3 = bitcast double* %1 to i8*
+;CHECK-NEXT:  tail call void @free(i8* %3)
+;CHECK-NEXT:  call void @cblas_dgemm(i32 102, i32 112, i32 111, i32 2, i32 3, i32 4, double %alpha, double* nocapture readonly %1, i32 4, double* nocapture readonly %"Cfoo'", i32 4, double 1.000000e+00, double* %"Bfoo'", i32 2)
+;CHECK-NEXT:  %4 = bitcast double* %2 to i8*
+;CHECK-NEXT:  tail call void @free(i8* %4)
+;CHECK-NEXT:  call void @cblas_dscal(i32 12, double %beta, double* %"Cfoo'", i32 1)
+;CHECK-NEXT:  ret { double, double } zeroinitializer
```

Comment:
Why is this zero? Shouldn't we also consider the case where alpha and beta are active?